src/solvers/flattening/boolbv_width.cpp

```diff
@@ -136,13 +135,7 @@ const boolbv_widtht::entryt &boolbv_widtht::get_entry(const typet &type) const
     else
     {
       mp_integer total = *array_size * sub_width;
-      if(total>(1<<30)) // realistic limit
-        throw analysis_exceptiont("array too large for flattening");
-
-      if(total < 0)
```

Comment:
This case does not seem to be handled any more.  Maybe it could become an invariant?

Reply:
I did some checking and this error handling should be left in since it can trigger an exception if not handled. I've pushed a fix here.